src/pybind/ceph_volume_client.py

```diff
@@ -1419,6 +1418,33 @@ def put_object(self, pool_name, object_name, data):
         finally:
             ioctx.close()
 
+    def put_object_versioned(self, pool_name, object_name, data, version):
```

Comment:
Because we still need a separate `get_object_versioned` method, I think we should also have a separate put method. The `put_object` method can simply call `put_object_versioned` and just pass `version=None`.